### PR TITLE
Review "Inner" snippets

### PR DESCRIPTION
**Pull request by rdeodoo**
Review the snippets which are in the "inner" section.

https://www.odoo.com/web#id=2152191&action=333&active_id=974&model=project.task&view_type=form&menu_id=4720

### CODE REVIEW

File: addons/website/static/src/snippets/s_hr/000.scss [hunk 1]
```diff
@@ -12,13 +12,4 @@
...
     .s_hr_solid  { border-top-style: solid;  }
-    // Thickness
-    $count: 5;
-    $base-width: 1px;
-    @while $count > 0 {
-        .s_hr_#{$base-width * $count} {
-            border-top-width: $base-width * $count;
-        }
-        $count: $count - 1;
-    }
```
Comment (qsm-odoo) [inline addons/website/static/src/snippets/s_hr/000.scss]: Incompatible change -> leave that here and create a new version of s_hr increasing the css compatibility count :)

File: addons/website/views/snippets/s_alert.xml [hunk 2]
```diff
@@ -12,19 +12,24 @@
 
 <template id="s_alert_options" inherit_id="website.snippet_options">
     <xpath expr="." position="inside">
-        <div data-selector=".s_alert">
+        <div data-selector=".s_alert" data-js="Alert">
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_alert.xml]: Why ?
EDIT: mmh in fact I think it is just your commit which is not split correctly, the Alert option should be in here
EDIT2: mmh in fact I cannot find that Alert option anywhere ?

File: addons/website/views/snippets/s_alert.xml [hunk 3]
```diff
@@ -12,19 +12,24 @@
...
-        </div>
-
-        <div data-selector=".s_alert">
             <we-select string="Size">
                 <we-button data-select-class="s_alert_sm">Small</we-button>
                 <we-button data-select-class="s_alert_md">Medium</we-button>
                 <we-button data-select-class="s_alert_lg">Large</we-button>
             </we-select>
+            <we-select string="Type" data-apply-to=".fa">
+                <we-button data-select-class="fa-user-circle" data-select-type="alert-primary">Primary</we-button>
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_alert.xml]: Reading this, I think that if someone puts icons inside an alert, changing the alert type will change all the icons he put. Probably need a specific class on the icon to change ?

File: addons/website/static/src/scss/website.scss [hunk 4]
```diff
@@ -199,6 +199,11 @@ font[class*='bg-'] {
         border-top-left-radius: $size;
         border-bottom-right-radius: $size;
     }
+    &.rounded-circle-empty {
+        @extend .rounded-circle;
+        border: 2px solid #dddddd;
```
Comment (qsm-odoo) [inline addons/website/static/src/scss/website.scss]: Probably need bootstrap variables instead

File: addons/website/static/src/snippets/s_share/000.scss [hunk 5]
```diff
@@ -10,16 +10,30 @@
...
+            .fa.rounded-circle-empty { border-color: #db4437; }
+        }
+        .s_share_whatsapp {
+            &, &:hover, &:focus { color: #25d366; }
+            .fa.rounded-circle-empty { border-color: #25d366; }
+        }
+        .s_share_pinterest {
+            &, &:hover, &:focus { color: #C8232C; }
+            .fa.rounded-circle-empty { border-color: #C8232C; }
+        }
```
Comment (qsm-odoo) [inline addons/website/static/src/snippets/s_share/000.scss]: Was like that before but if those are social media colors, there are variables for them :+1: 
Also, is this useful to target `.rounded-circle-empty` ? Maybe just .fa is enough ?

File: addons/website/static/src/scss/website.scss [hunk 6]
```diff
@@ -199,6 +199,11 @@ font[class*='bg-'] {
         border-top-left-radius: $size;
         border-bottom-right-radius: $size;
     }
+    &.rounded-circle-empty {
```
Comment (qsm-odoo) [inline addons/website/static/src/scss/website.scss]: Maybe rounded-empty-circle? :3

File: addons/website/views/snippets/s_share.xml [hunk 7]
```diff
@@ -2,7 +2,7 @@
 <odoo>
 
 <template id="s_share" name="Share">
-    <div t-attf-class="s_share #{_classes}">
+    <div t-attf-class="s_share text-left #{_classes}">
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_share.xml]: Why is this needed ?

File: addons/website_mail_channel/static/src/js/website_mail_channel.editor.js [hunk 8]
```diff
@@ -8,32 +8,68 @@ var wUtils = require('website.utils');
 var _t = core._t;
 
 options.registry.subscribe = options.Class.extend({
-    select_mailing_list: function (previewMode, value) {
+    /**
+     * @override
+     */
+    cleanForSave: function () {
+        this.$target.addClass('d-none');
+    },
```
Comment (qsm-odoo) [inline addons/website_mail_channel/static/src/js/website_mail_channel.editor.js]: I know you just moved it but is this even needed? Is not there a related public widget whose destroy is doing this?

File: addons/website_mail_channel/views/snippets.xml [hunk 9]
```diff
@@ -16,7 +16,7 @@
             <div class="oe_snippet_body js_follow_alias"
                       data-id="0"
                       data-object="mail.channel"
-                      data-follow="off">
+                      data-follow="off" data-name="Discussion Group">
```
Comment (qsm-odoo) [inline addons/website_mail_channel/views/snippets.xml]: Maybe a great time to make an actual snippet out of this ? (a view with a dedication folder + options etc) ? So that this is not needed. (Maybe do that in a separate commit so that I can try keeping the history the day I merge it).

File: addons/website/static/src/snippets/s_share/000.scss (hunk 5 above)
Comment (rdeodoo) [reply addons/website/static/src/snippets/s_share/000.scss]: Changed and used variables, added missing ones.
Yes it is mandatory, as the 3 different layout just change the `rounded-circle-empty` to `rounded-circle` or nothing but the `fa` always remain.

File: addons/website/views/snippets/s_alert.xml (hunk 2 above)
Comment (rdeodoo) [reply addons/website/views/snippets/s_alert.xml]: Weird, I find the code back through `reflog`, you might have another look, the options JS is now there.

File: addons/website/views/snippets/s_alert.xml (hunk 3 above)
Comment (rdeodoo) [reply addons/website/views/snippets/s_alert.xml]: Good catch, done

File: addons/website/views/snippets/s_share.xml (hunk 7 above)
Comment (rdeodoo) [reply addons/website/views/snippets/s_share.xml]: We now have a new snippet option for s_share to center left/center/right.
Defaulting to left

File: addons/website_mail_channel/static/src/js/website_mail_channel.editor.js (hunk 8 above)
Comment (rdeodoo) [reply addons/website_mail_channel/static/src/js/website_mail_channel.editor.js]: Not right now, it is just the public widget start [removing the d-none](https://github.com/odoo/odoo/blame/193040ada2cd8e94a04383bbfea7f56a1a19340e/addons/website_mail_channel/static/src/js/website_mail_channel.snippet.js#L22)
I guess we should refactor it..

File: addons/website_mail_channel/views/snippets.xml (hunk 9 above)
Comment (rdeodoo) [reply addons/website_mail_channel/views/snippets.xml]: Alright, I will add it on top of the current PR

File: addons/website/static/src/scss/website.scss (hunk 4 above)
Comment (rdeodoo) [reply addons/website/static/src/scss/website.scss]: For the color? Done

File: addons/website/static/src/snippets/s_share/000.scss (hunk 5 above)
Comment (qsm-odoo) [reply addons/website/static/src/snippets/s_share/000.scss]: Yes, but do they have a border ? What I mean is that if you had `.fa { border-color: XXX; }` on an element without border, it is fine.

**Comment by qsm-odoo** [comment]
https://github.com/odoo/odoo/pull/42387#discussion_r361499202 -> maybe even for the border: $border-width ?

File: addons/website/views/snippets/s_alert.xml [hunk 10]
```diff
@@ -12,19 +12,24 @@
...
                 <we-button data-select-class="s_alert_lg">Large</we-button>
             </we-select>
+            <we-select string="Type" data-apply-to=".fa.s_alert_icon">
+                <we-button data-select-class="fa-user-circle" data-select-type="alert-primary">Primary</we-button>
+                <we-button data-select-class="fa-user-circle-o" data-select-type="alert-secondary">Secondary</we-button>
+                <we-button data-select-class="fa-info-circle" data-select-type="alert-info">Info</we-button>
+                <we-button data-select-class="fa-check-circle" data-select-type="alert-success">Success</we-button>
+                <we-button data-select-class="fa-exclamation-triangle" data-select-type="alert-warning">Warning</we-button>
+                <we-button data-select-class="fa-exclamation-circle" data-select-type="alert-danger">Danger</we-button>
+            </we-select>
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_alert.xml]: We discussed with SBU, reorganize the options to have type, width, size, color in that order.

File: addons/website/static/src/snippets/s_alert/options.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    selectType: function (previewMode, widgetValue, params) {
+        const $snippet = this.$target.parent('.s_alert');
+        $snippet.removeClass('alert-delta'); // TODO (review) - migration/compatibility
+        $snippet.removeClass(params.possibleValues.join(' '));
+        $snippet.addClass(widgetValue);
```
Comment (qsm-odoo) [inline addons/website/static/src/snippets/s_alert/options.js]: Mmmh... all of this is kinda strange :/ You have to find back the widget because you used an apply-to, you must reimplement selectClass for the alert- stuff etc :/ and this cannot even work since you don't know the colors to remove (your TODO).

Note: this also revealed a bug I will fix in stable versions as well (the info/success/... colors cannot be removed).

Maybe the best is to do a hack for now: search for the colorpicker widget in your option DOM and click on the appropriate color by hand when selecting a specific type ? (add a comment explaining)

Comment (rdeodoo) [reply addons/website/static/src/snippets/s_alert/options.js]: I am not sure to fully understand the issue but this is working as I expect it to?
When you select one of the available preset type (danger, warning..) it will use the related color.
If you want another one, using the colorpicker will prevail on the preset type, then you can switch type without it changing the color as you selected a specific color ?
So specific color > layout color, but you can still change layout to switch icon etc.

> this cannot even work since you don't know the colors to remove (your TODO).

No, the TODO is just for compatibility with older snippet which have the `alert-delta` class that I removed, just that line makes it compatible (otherwise it would not work and we would need a new snippet version)

And I don't think it is weird to find the parent? I need to toggle classes on 2 different DOM elements with the same option, rather use the native select-class for one and a method for the other, no?

File: addons/website/views/snippets/s_alert.xml (hunk 10 above)
Comment (rdeodoo) [reply addons/website/views/snippets/s_alert.xml]: If you want that exact order I'll have to split the options in 2 xpath to have some before and some after generic options Width (not specific to s_alert) :nauseated_face: 
Let's go I guess

File: addons/website/static/src/snippets/s_share/000.scss (hunk 5 above)
Comment (rdeodoo) [reply addons/website/static/src/snippets/s_share/000.scss]: They don't, until they do (theme or whatever). I don't think targeting `.fa.rounded-circle-empty` with just `.fa` as the other `fa` elements have no border is really what we should do, it will work by chance for now until it won't

File: addons/website/static/src/scss/website.scss (hunk 4 above)
Comment (rdeodoo) [reply addons/website/static/src/scss/website.scss]:
```scss
border: (2 * $border-width) solid $gray-400;
```
:+1:

**Comment by rdeodoo** [comment]
@qsm-odoo Most have been processed. A few left unresolved, you might wanna have a look at those.

File: addons/website/static/src/snippets/s_share/000.scss (hunk 5 above)
Comment (qsm-odoo) [reply addons/website/static/src/snippets/s_share/000.scss]: I would only use `.fa` as it allow to add custom style without having to implement the border-color. But in fact, even better: I think you don't need those rules at all and the border should be automatically of the right color by changing your rule https://github.com/odoo/odoo/pull/42387/files#diff-a50ffff96791f36c6dbe5c9273135e5aR202 and not set the border-color.

File: addons/website/views/snippets/s_alert.xml (hunk 10 above)
Comment (qsm-odoo) [reply addons/website/views/snippets/s_alert.xml]: Mmh... too bad but yes "type" should really be the one the users use first and color as a last resort.

File: addons/website/static/src/snippets/s_alert/options.js (hunk 11 above)
Comment (qsm-odoo) [reply addons/website/static/src/snippets/s_alert/options.js]: Personally, when you select the type, I would clearly want the color to change back again (and that's why type should be first). The status color do not even appears in the colorpicker :/ Personnaly, what I would do as an user is:
0. drop the alert snippet
1. add the alert text
2. select the alert type
3. try another color just for fun
4. trying to revert back to the correct type color... which I can't without doing all over again.

No? @sbuhl ?

Comment (rdeodoo) [reply addons/website/static/src/snippets/s_alert/options.js]: You just need to reset the colorpicker with the reset button

I'd type my text, select the color I am fine with (probably a light blue/gray) and then change type to see the icons but I really don't think there is a 'normal' flow for this one..

That'd as much looks buggy to me if it would change the color I just chose when I change the type of the alert (which is not a color but multiple set up, color and icon in this case).

I don't care tbh or it will never end :zzz:

File: addons/website/static/src/scss/website.scss [hunk 12]
```diff
@@ -199,6 +199,11 @@ font[class*='bg-'] {
         border-top-left-radius: $size;
         border-bottom-right-radius: $size;
     }
+    &.rounded-empty-circle {
+        @extend .rounded-circle;
+        border: (2 * $border-width) solid $gray-400;
```
Comment (qsm-odoo) [inline addons/website/static/src/scss/website.scss]: @rdeodoo Maybe a bit too violent? Maybe `ceil(1.4 * $border-width)`? Or am I overthinking this?

File: addons/website/static/src/snippets/s_alert/options.js (hunk 11 above)
Comment (rdeodoo) [reply addons/website/static/src/snippets/s_alert/options.js]: Well actually I do care, I think losing the color I chose and I wont be able to find back just because I wanted to select another icon is not cool.
How I implemented it won't lose anything, it's just the selected color that prevail but you can unset it through the colorpicker if you don't want it anymore.

File: addons/website/static/src/scss/website.scss (hunk 12 above)
Comment (rdeodoo) [reply addons/website/static/src/scss/website.scss]: As you prefer, I don't see what was wrong with `2px` anyway as whatever theme you use you most likely won't want a thinner/bigger border in this case, so I might as well do that.

Screenshot from 2019-12-27 16-01-23

File: addons/website/static/src/snippets/s_share/000.scss (hunk 5 above)
Comment (rdeodoo) [reply addons/website/static/src/snippets/s_share/000.scss]: Good catch, we don't need that rule actually so there is no debate on the `.fa.selector` anymore I guess :white_flag:

File: addons/website/static/src/snippets/s_alert/options.js (hunk 11 above)
Comment (qsm-odoo) [reply addons/website/static/src/snippets/s_alert/options.js]: Ah ah I prefer when you care :p I think this is important if we want to make the UI perfect. I suppose it is a matter of opinion but I don't agree with you here. It is the same with other options, you set a color, then set a background (below option), if you want to change the color again, your background is removed. Otherwise you do not have any apparent effect for the color, which is the main option.

In your case, why would you be ok to lose the icon when you change the type but not ok to lose the color (to match the type). Also you lose it only if you click on a type, you can also not choose anything after previewing them.

Anyways, I'll @sbuhl decide :+1:

File: addons/website/static/src/scss/website.scss (hunk 12 above)
Comment (qsm-odoo) [reply addons/website/static/src/scss/website.scss]: Mmh, maybe you are right, border-width are often difficult to have an opinion on... but if a theme is stupid enough to have border-width: 5px... I think the logic would be to use bigger width here too, no? but as the size is also hardcoded... don't know.

File: addons/website/static/src/snippets/s_alert/options.js (hunk 11 above)
Comment (rdeodoo) [reply addons/website/static/src/snippets/s_alert/options.js]: Despite the fact that the behavior would change if using colorpicker theme color or using other colors, I stay on my position where I'd not want my so-hardly-chosen color to be erased (even in preview mode) when changing type (even if DOM not 100% bootstrap compliant in some cases with theme color). Indeed, I change the **type/display/layout**, not the color, which is only a part of the type/display/layout.

But as long discussed today, I don't think we will come to a consensus anytime soon. And since it has no value whatsoever to implement one or the other solution (the snippet value would probably increase/decrease by 0.001% depending of your position), let's do what you propose so we move forward:

Changing **layout** will erase the chosen color.

Note that I left a TODO in it while your fix/imp is not merged yet. Without it it won't work except for 2 colors.

File: addons/website/views/snippets/s_share.xml [hunk 13]
```diff
@@ -13,12 +13,38 @@
...
+                <we-button data-select-class="">Left</we-button>
+                <we-button data-select-class="d-none">None</we-button>
+            </we-select>
+            <we-select string="Icon Layout" data-apply-to=".fa">
+                <we-button data-select-class="rounded shadow-sm">Square</we-button>
+                <we-button data-select-class="rounded-empty-circle shadow-sm">Circle</we-button>
+                <we-button data-select-class="rounded-circle shadow-sm">Disk</we-button>
+                <we-button data-select-class="m-3">None</we-button>
+            </we-select>
+            <!-- Compatibility, keep reverse logical, don't use `icon_color` -->
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_share.xml]: I am fine with keeping one mention about compatibility (even if I don't like it) but choose: xml or scss :p

File: addons/website_mail_channel/static/src/snippets/s_channel/000.js [hunk 14]
```diff
@@ -102,22 +98,48 @@ publicWidget.registry.follow_alias = publicWidget.Widget.extend({
...
+
+    /**
+     * @private
+     */
+    _onFollowBtnClick: function (ev) {
+        if ($(ev.currentTarget).closest('.js_mg_follow_form').length) {
+            this.$('.js_follow_email').val($(ev.currentTarget).closest('.js_mg_follow_form').find('.js_follow_email').val());
+        }
+    },
+    _onFollowUnFollowBtnClick: function (ev) {
```
Comment (qsm-odoo) [inline addons/website_mail_channel/static/src/snippets/s_channel/000.js]: `@private`

File: addons/website_mail_channel/static/src/snippets/s_channel/options.js [hunk 15]
```diff
@@ -0,0 +1,75 @@
+odoo.define('website_mail_channel.editor', function (require) {
```
Comment (qsm-odoo) [inline addons/website_mail_channel/static/src/snippets/s_channel/options.js]: Maybe rename the module (also 000.js one) with our convention?

File: addons/website_mail_channel/views/snippets.xml [hunk 16]
```diff
@@ -1,52 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<odoo>
-
-<template id="remove_external_snippets" inherit_id="website.external_snippets">
-    <xpath expr="//t[@t-install='website_mail_channel']" position="replace"/>
-</template>
-
```
Comment (qsm-odoo) [inline addons/website_mail_channel/views/snippets.xml]: Actually this file should have stayed to handle snippet removal and addition ;)

File: addons/website_mail_channel/views/snippets/s_channel.xml [hunk 17]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+<template id="s_channel" name="Discussion Group">
+    <div class="oe_snippet_body js_follow_alias"
```
Comment (qsm-odoo) [inline addons/website_mail_channel/views/snippets/s_channel.xml]: You can remove oe_snippet_body, this is an automatic class added when using the system correctly ;)

File: addons/website/static/src/snippets/s_alert/options.js [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    selectType: function (previewMode, widgetValue, params) {
+        // compatibility: we need `alert-delta` to be removed, which will be the
+        // case by selecting another color with the colorpicker.
```
[the remark this answers is not in the document]
Comment (rdeodoo) [reply addons/website/static/src/snippets/s_alert/options.js]: Yep but since the colorpicker was not removing alert-xxx, I thought it was normal behavior, so it was important to remove alert-delta. This is not relevant anymore indeed.

File: addons/website/static/src/snippets/s_alert/options.js [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+        let widget = null;
+        // TODO (review) This will not work for every color, need QSM fix/imp?
```
[the remark this answers is not in the document]
Comment (rdeodoo) [reply addons/website/static/src/snippets/s_alert/options.js]: Indeed :+1:

File: addons/website/static/src/snippets/s_alert/options.js [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+        this.trigger_up('user_value_widget_request', {
+            name: 's_alert_colorpicker',
+            onSuccess: _widget => widget = _widget,
+        });
+        const eventName = previewMode === 'reset' ? 'mouseleave' : previewMode ? 'mouseenter' : 'click';
+        widget.$el.find(`[data-color="${widgetValue}"]`).trigger(eventName);
```
[the remark this answers is not in the document]
Comment (rdeodoo) [reply addons/website/static/src/snippets/s_alert/options.js]: Well the better way gotta be find now, this is not working anymore since I rebased Tuesday :zzz: 
The select is not closing anymore when clicked because of this line 
```javascript
widget.$el.find(`[data-color="${widgetValue}"]`).trigger(eventName);
```
Any hints? Probably something deployed recently

File: addons/website_mail_channel/views/snippets.xml (hunk 16 above)
Comment (rdeodoo) [reply addons/website_mail_channel/views/snippets.xml]: Not sure I understand this one

File: addons/website/static/src/snippets/s_alert/options.js (hunk 20 above)
[the remark this answers is not in the document]
Comment (qsm-odoo) [reply addons/website/static/src/snippets/s_alert/options.js]: Don't know, I'll test

File: addons/website_mail_channel/views/snippets.xml (hunk 16 above)
Comment (qsm-odoo) [reply addons/website_mail_channel/views/snippets.xml]: t-install and t-snippet should not have been moved in the dedicated snippet xml file but should have stayed here.

Comment (rdeodoo) [reply addons/website_mail_channel/views/snippets.xml]: Hm, thought u meant it was an error.
Alright, t-install and t-snippet should be inside `/views/snippets/snippets.xml`.
That file did not exists inside `website_mail_channel`, created it, should be good now

File: addons/website/static/src/snippets/s_alert/options.js (hunk 20 above)
[the remark this answers is not in the document]
Comment (rdeodoo) [reply addons/website/static/src/snippets/s_alert/options.js]: Not needed anymore with the `data-trigger` imp

**Comment by rdeodoo** [comment]
Rebased on top of https://github.com/odoo/odoo/pull/42986 which was just merged, removed the cherry picked commit of that PR

File: addons/website/views/snippets/s_alert.xml [hunk 21]
```diff
@@ -11,20 +11,30 @@
...
-    <xpath expr="." position="inside">
-        <div data-selector=".s_alert">
-            <we-colorpicker string="Color"
-                data-select-style="true"
-                data-css-property="background-color"
-                data-color-prefix="alert-"/>
+    <xpath expr="//div[@id='so_width']" position="before">
+        <div data-selector=".s_alert" data-js="Alert">
+            <we-select string="Type" data-apply-to=".fa.s_alert_icon">
+                <we-button data-select-class="fa-user-circle" data-trigger="s_alert_colorpicker" data-trigger-value="primary">Primary</we-button>
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_alert.xml]: You can put the `data-trigger` on the `we-select` if I implemented this correctly. So that you won't have to repeat it for each button.

File: addons/website/views/snippets/s_alert.xml [hunk 22]
```diff
@@ -11,20 +11,30 @@
...
+                <we-button data-select-class="fa-info-circle" data-trigger="s_alert_colorpicker" data-trigger-value="info">Info</we-button>
+                <we-button data-select-class="fa-check-circle" data-trigger="s_alert_colorpicker" data-trigger-value="success">Success</we-button>
+                <we-button data-select-class="fa-exclamation-triangle" data-trigger="s_alert_colorpicker" data-trigger-value="warning">Warning</we-button>
+                <we-button data-select-class="fa-exclamation-circle" data-trigger="s_alert_colorpicker" data-trigger-value="danger">Danger</we-button>
+            </we-select>
         </div>
-
+    </xpath>
+    <!-- Keep those options in separate xpath for options order -->
+    <xpath expr="//div[@id='so_width']" position="after">
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_alert.xml]: I don't see this `so_width`, you probably added it in the wrong commit... or maybe it is in a commit which is before in the tree but not in the PR order? Could you also name it differently? It will be confusing with the regular/small/full option which is also named width.

File: addons/website_mail_channel/views/snippets.xml [hunk 23]
```diff
@@ -44,7 +44,13 @@
...
              data-selector=".js_follow_alias"
              data-drop-near="p, h1, h2, h3, blockquote, .card">
-            <we-button data-select_mailing_list="" data-no-preview="true">Change Discussion List</we-button>
+            <we-row>
+                <we-select class="select_discussion_list" data-attribute-name="id" data-no-preview="true">
+                    <!-- 'we-button' added programmatically with DB data -->
+                </we-select>
+                <we-button class="fa fa-fw fa-plus" title="Create a public discussion group in your backend"
+                           data-create-channel="" data-no-preview="true"/>
+            </we-row>
```
Comment (qsm-odoo) [inline addons/website_mail_channel/views/snippets.xml]: There is no label for the select/row ?

Comment (rdeodoo) [reply addons/website_mail_channel/views/snippets.xml]: Nop, by design/spec, was discussed with sbu.
Does the UX seems weird to you or is it just codewise?

PS: Another `title` attribute :D

File: addons/website/views/snippets/s_alert.xml (hunk 22 above)
Comment (rdeodoo) [reply addons/website/views/snippets/s_alert.xml]: Yes added in a separate commit, where `so_text_align` is also added on Alignment option which is adding text-center class etc.

Hm, this one is also named `Width`, thus the `so_width`.

Imo `so_width` is fine for this one, which actually play with the width (w-25 w-50..), the ID describe exactly what the option does.
regular/small/full should have `so_container_width` if we need to add an ID one day, no?

Up to ID suggestion otherwise

File: addons/website/views/snippets/s_alert.xml (hunk 21 above)
Comment (rdeodoo) [reply addons/website/views/snippets/s_alert.xml]: Indeed :+1: well implemented :smile_cat:

File: addons/website_mail_channel/views/snippets.xml (hunk 23 above)
Comment (qsm-odoo) [reply addons/website_mail_channel/views/snippets.xml]: I did not look at the UX ;) It was just to be sure it was not an oversight

File: addons/website/views/snippets/s_alert.xml (hunk 22 above)
Comment (qsm-odoo) [reply addons/website/views/snippets/s_alert.xml]: Yep, fine. My concern was mainly about commit order here but did not check.

File: addons/website/static/src/snippets/s_hr/001.scss [hunk 24]
```diff
@@ -0,0 +1,15 @@
...
+        padding: 0;
+        border: 0;
+        border-top-color: inherit;
+        margin: 0;
+    }
+    // Style
+    .s_hr_dashed { border-top-style: dashed; }
+    .s_hr_double { border-top-style: double; }
+    .s_hr_dotted { border-top-style: dotted; }
+    .s_hr_solid  { border-top-style: solid;  }
```
Comment (qsm-odoo) [inline addons/website/static/src/snippets/s_hr/001.scss]: @rdeodoo In fact, since you are increasing the CSS counter for a reason, I will remove those as well, less CSS :+1: Don't worry, I am doing it :)

File: addons/website_mail_channel/static/src/snippets/s_channel/000.js [hunk 25]
```diff
@@ -1,11 +1,19 @@
-odoo.define('website_mail_channel.snippet', function (require) {
```
Comment (qsm-odoo) [inline addons/website_mail_channel/static/src/snippets/s_channel/000.js]: @rdeodoo Note: for files you split, there is nothing you can do since it requires a merge I may or not do by myself during one night. But for files you move then slightly modify, you should make the move in a separate commit. Here GitHub is intelligent enough to know it is a move but our forward-port bot won't be ;) I'll do it by myself here since already cherry-picking ;)

**Comment by qsm-odoo** [comment]
Merged this in main branch: https://github.com/odoo/odoo/pull/42937
